### PR TITLE
mes-1014 remove driving faults and mes-1924 remove serious and dangerous faults

### PR DESCRIPTION
**Pull request by Jason-Pullen**
## Description and relevant Jira numbers
Test report now has the ability to remove and add Driving, Serious and Dangerous faults
## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/pages/test-report/components/competency/__tests__/competency.spec.ts [hunk 1]
```diff
@@ -89,139 +92,320 @@ describe('CompetencyComponent', () => {
...
+      it('should dispatch a ADD_DANGEROUS_FAULT action if dangerous mode is active on press and hold', () => {
+        component.competency = Competencies.clearance;
+        component.isDangerousMode = true;
+
+        const storeDispatchSpy = spyOn(store$, 'dispatch');
+        component.addOrRemoveFault(false);
+
+        expect(storeDispatchSpy).toHaveBeenCalledWith(new AddDangerousFault(component.competency));
+        expect(storeDispatchSpy).toHaveBeenCalledWith(new ToggleDangerousFaultMode());
+      });
```
Comment (rossfauldsbjss) [inline src/pages/test-report/components/competency/__tests__/competency.spec.ts]: add dangerous fault is active on press, and add dangerous fault is active on press and hold are identical.
So - one of them isn't doing what it says in the description or one is unnnecessary

File: src/pages/test-report/components/competency/__tests__/competency.spec.ts [hunk 2]
```diff
@@ -89,139 +92,320 @@ describe('CompetencyComponent', () => {
...
+        component.hasDangerousFault = true;
+        component.isDangerousMode = true;
+
+        const storeDispatchSpy = spyOn(store$, 'dispatch');
+        component.addOrRemoveFault(true);
+
+        expect(storeDispatchSpy).not.toHaveBeenCalledWith(new AddDangerousFault(component.competency));
+        expect(storeDispatchSpy).not.toHaveBeenCalledWith(new ToggleDangerousFaultMode());
+      });
+    });
```
Comment (rossfauldsbjss) [inline src/pages/test-report/components/competency/__tests__/competency.spec.ts]: duplicated test for ADD_DANGEROUS_FAULT action if there is a dangerous fault

Comment (Jason-Pullen) [reply src/pages/test-report/components/competency/__tests__/competency.spec.ts]: yup looks that way.. nice spot

File: src/pages/test-report/components/competency/__tests__/competency.spec.ts (hunk 1 above)
Comment (Jason-Pullen) [reply src/pages/test-report/components/competency/__tests__/competency.spec.ts]: component.addOrRemoveFault(false); boolean flipped to force press or press and hold path

**Pull request merged**